src/Voron/Impl/LowLevelTransaction.cs

```diff
@@ -940,8 +940,8 @@ public void EndAsyncCommit()
             if (AsyncCommit.Result)
                 Environment.LastWorkTime = DateTime.UtcNow;
 
-            CommitStage3_DisposeTransactionResources();
             BeforeCommitFinalization?.Invoke(this);
+            CommitStage3_DisposeTransactionResources();
```

Comment:
This shouldn't be moved, right?

Reply:
It should to be aligned with regular commit. When moving this for commit it broke things pretty badly and async commit have the order backwards. This leads me to believe there is a very rare case where async commit will fail horribly.

Reply:
It isn't _actually_ a problem, because there is a lock on the _next_ transaction, but that is only true right now and is a bug waiting to happen. 
We're better of _with_ this change.